pkg/apis/v1alpha1/subnet_types.go

```diff
@@ -0,0 +1,78 @@
+/* Copyright © 2021 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type SubnetStatusCondition string
+
+const (
+	SubnetReady    SubnetStatusCondition = "Ready"
+	SubnetNotReady SubnetStatusCondition = "NotReady"
+)
+
+// SubnetSpec defines the desired state of Subnet.
+type SubnetSpec struct {
+	// Size of Subnet based upon estimated workload count.
+	// Defaults to 64.
+	// +kubebuilder:default:=64
+	IPv4SubnetSize int `json:"ipv4SubnetSize,omitempty"`
+	// Access mode of Subnet, accessible only from within VPC or from outside VPC.
+	// Defaults to private.
+	// +kubebuilder:default:=private
+	// +kubebuilder:validation:Enum=private;public;isolated
+	AccessMode string `json:"accessMode,omitempty"`
+	// Subnet CIDR.
+	// +kubebuilder:validation:MinItems=0
+	// +kubebuilder:validation:MaxItems=2
+	IPAddresses []string `json:"ipAddresses,omitempty"`
+}
+
+// SubnetCondition defines condition of Subnet.
+type SubnetCondition struct {
+	// Type defines condition type.
+	Type SubnetStatusCondition `json:"type"`
+	// Status defines status of condition type, True or False.
+	Status corev1.ConditionStatus `json:"status"`
+	// Reason shows a brief reason of condition.
+	Reason string `json:"reason,omitempty"`
+	// Message shows a human readable message about condition.
+	Message string `json:"message,omitempty"`
+}
+
+// SubnetStatus defines the observed state of Subnet.
+type SubnetStatus struct {
+	// Logical switch ID.
+	LsID       string            `json:"lsID"`
+	SubnetCIDR string            `json:"subnetCIDR"`
```

Comment:
based on above , shall we also change to ipAddresses with slice type?

Reply:
Thanks and done.